hstream/src/HStream/Server/Core/Subscription.hs

```diff
@@ -236,13 +236,20 @@ streamingFetchCore ctx SFetchCoreDirect = \initReq callbacksGen -> do
         return $ Right ()
   consumerCtx <- initConsumer scwContext (streamingFetchRequestConsumerName initReq) Nothing Nothing streamSend
   Log.debug "pass initConsumer"
-  let streamRecv = do
-        req <- atomically $ readTChan mockAckPool
-        return $ Right (Just req)
-  withAsync (recvAcks ctx scwState scwContext consumerCtx streamRecv) wait `onException` do
-    case tid_m of
-      Just tid -> killThread tid
-      Nothing  -> return ()
+  let streamRecv' =
+        (do isClosed <- readTVar consumerClosed
+            check isClosed
+            return (Right Nothing))
+        `orElse`
+        (tryReadTChan mockAckPool >>= \case
+            Nothing -> retry
+            req     -> return $ Right req)
+  let streamRecv = atomically streamRecv'
+  withAsync (recvAcks ctx scwState scwContext consumerCtx streamRecv) wait
+  -- `onException` do
+  --   case tid_m of
+  --     Just tid -> killThread tid
+  --     Nothing  -> return ()
```

Comment:
The `StreamReadClose` exception will be thrown to the upper layer now. However, this can bring a strange error message while pressing `C-c` in the cli terminal. This is because the cli session invokes `streamingFetchCfore` to get results from the sink stream. The error message looks like

```
[FATAL][2023-04-26T08:51:39+0000][./HStream/Exception.hs:790:7][thread#57]StreamReadClose_ {error:ErrorCodeInternalError,message:StreamReadClose,extra:"Consumer is closed"}
CallStack (from HasCallStack):
  StreamReadClose, called at src/HStream/Server/Core/Subscription.hs:892:21 in hstream-0.1.0.0-inplace:HStream.Server.Core.Subscription
```